src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -426,19 +427,37 @@ private static void makePublic(final CtClass c) {
         boolean foundDefault = false;
         if (!cl.isInterface()) {
           for (CtConstructor ctor : cl.getConstructors()) {
-            if (ctor.getParameterTypes().length == 0
-                && (ctor.getModifiers() & javassist.Modifier.ABSTRACT) == 0
-                && !ctor.isEmpty()) {
+            if (ctor.isEmpty()
```

Comment:
>public boolean isEmpty​()
>Returns true if the constructor (or static initializer) is the default one. This method returns true if the constructor takes some arguments but it does not perform anything except calling super() (the no-argument constructor of the super class).

Why are we completely discarding the case where the constructor takes no arguments here? Wouldn't both cases be possible, i.e. should this line be `if ((ctor.isEmpty || ctor.getParameterTypes().length == 0)`?

Reply:
you're right, I had missed the second part of the description